### DISCUSSION TITLE
What can I use in macOS to show an Emacs GUI frame ready to call consult-omni?

### DISCUSSION THREAD

**Discussion by ghost** [armindarvish/consult-omni#20] [Q&A]
Do I have to use Emacsclient and open that with an assigned shortcut or how does it work? Do you have some configuration for the looks of the buffer that opens, so that it is customized for consult-omni search?

I've gone through 2 hours of the video1, and I still need to see the 1.5 hours, but maybe it's not something documented/shown in the video (last part is about extending source code, I think).

**Comment by armindarvish** [comment]
@oscarvarto I show how to get the desktop integration working right here: https://www.youtube.com/watch?v=wNH2E7iT__c&t=8593s

It covers the general case and yes, it does require an emacsclient.


Personally, I use [yequake](https://github.com/alphapapa/yequake). Here is an example for yequake for MacOS (on other OS you likely need to change the `window-system`).

```emacs-lisp
(add-to-list 'yequake-frames '("consult-omni-demo"
                               (buffer-fns . #'consult-omni-multi)
                               (width . 0.8)
                               (height . 0.8)
                               (top . 0)
                               (frame-parameters . ((name . "yequake-demo")
                                                    (minibuffer . t)
                                                    (autoraise . t)
                                                    (window-system . ns) ;;change accordingly
                                                    ))))
```

Then to call the frame:

```emacs-lisp
emacsclient --eval "(yequake-toggle \"consult-omni-demo\")"
```

**Comment by ghost** [comment]
Hey @armindarvish. Thank you very much for the answer. I really appreciate the guidance.
The pointer you gave me to the video is at 2 hours 23 mins (of almost 3:30 hours!!). While having the video available, it is really, really long, and maybe it will be really complicated to find answers in a video such long (just my opinion). And believe me, I did watch 2 hours of the video1 during my vacations. I still don't watch the rest of it (I'll do it when I get more time available), and I am very thankful that you have taken the time to provide guidance somehow.

Still, I believe it would be much better to have more written. 3:30 hours of a video is way too much!

**Reply by ghost** [reply]
Maybe having an index with the subject and the time at which that subject is covered in the video is sort of a compromise that could  help take advantage of the great video without discarding it. I recognize that getting to view a video has its own advantages.

Other option might be to have a series of videos, each one with a title that makes it easier to find quickly the subject of interest, while making much more probable that people devotes time to smaller chunks of information.

**Comment by armindarvish** [comment]
@oscarvarto Thanks for feedback! 

For you and anyone else that has a similar opinion, here is a general response that I put here so I can refer people to it later:

1- The video is not part of the package, nor is the specific feature you are asking about (in this case, making new frames with emacsclient). These are nice-to-haves that I have in my setup, but not part of the package. The same that the theme or key bindings I use in the screenshots are not part of this package! Keep in mind that most other packages don't provide such nice-to-have video guidance, screenshots, ... 

2- Most advanced users (who are the likely users of a package like this) don't need such video instructions. The documents in the package should be enough for using it. If some users cannot figure it out from the documents, then they can refer to the video instructions and while that is 3.5 hours long, there are links to subsections in the video description on YouTube! So any user should be able to find the information they need. How much work it takes for them to find it is up to them!

3- Most importantly, this is an open-source free software that I do as a hobby on the side. I am not charging people for it, nor am I shoving advertisement or self-promotion at people for using it. So, unless somebody is sponsoring me or actively contributing to the package, etc. they should have zero expectations for support or help, let alone complaining about how long the video instruction is!

**Reply by ghost** [reply]
Sorry if my comments sounded like a complaint. That's not the case. It's an observation because I'm really interested in your work. I used almost a complete day to start with consult-omni (yeah, I'm not a power user), and the last part of that night watching 2 hours of the video1.

Thanks for the video, screenshots and all the docs you have provided to the community for free. Really, I have spent time with these resources, and finding answers on my own in the source code, etc. My comments were just that, an opinion/feedback. Please do not take it personal.

**Reply by ghost** [reply]
I didn't realize there was already an index in the video1. Thanks.

**Reply by armindarvish** [reply]
@oscarvarto I did not take the comments personally, nor I meant to suggest that you are complaining. 

I simply wrote that answer in a more general format, so I can refer other people to it in the future because I do expect to get "Oh why is the video so long comments in the future! 🫣" and so it helps to have the answer ready somewhere and use the link instead of typing it out every time.
